src/components/application_manager/test/message_helper/message_helper_test.cc

```diff
@@ -1058,6 +1058,41 @@ TEST_F(MessageHelperTest,
   }
 }
 
+TEST_F(MessageHelperTest, ProcessSoftButtons_TextAndImageExists_SUCCESS) {
+  std::vector<mobile_apis::SoftButtonType::eType> button_types_list;
+  button_types_list.push_back(mobile_apis::SoftButtonType::SBT_BOTH);
+  button_types_list.push_back(mobile_apis::SoftButtonType::SBT_IMAGE);
+  button_types_list.push_back(mobile_apis::SoftButtonType::SBT_TEXT);
+  std::vector<mobile_apis::SoftButtonType::eType>::iterator it =
+      button_types_list.begin();
+  smart_objects::SmartObject button_image;
+  button_image[strings::image_type] = mobile_apis::ImageType::STATIC;
+  const std::string img_file = "./file.txt";
+  button_image[strings::value] = img_file;
+  smart_objects::SmartObject soft_button;
+  soft_button[strings::system_action] = mobile_apis::SystemAction::INVALID_ENUM;
+  smart_objects::SmartObject message_params;
+  const std::string button_text = "soft_button.text";
+  MockApplicationSharedPtr app_shared_ptr =
+      utils::MakeShared<MockApplication>();
+  policy_test::MockPolicyHandlerInterface policy_handler;
+  application_manager_test::MockApplicationManager mock_application_manager;
+  for (; it != button_types_list.end(); ++it) {
+    soft_button[strings::type] = (*it);
+    message_params[strings::soft_buttons][0] = soft_button;
+    message_params[strings::soft_buttons][0][strings::text] = button_text;
+    message_params[strings::soft_buttons][0][strings::image] = button_image;
+    EXPECT_CALL(policy_handler, PolicyEnabled()).WillOnce(Return(true));
+    EXPECT_CALL(policy_handler, CheckSystemAction(_, _)).WillOnce(Return(true));
+    EXPECT_CALL(*app_shared_ptr, policy_app_id()).WillOnce(Return(""));
+    EXPECT_EQ(mobile_apis::Result::SUCCESS,
+              MessageHelper::ProcessSoftButtons(message_params,
+                                                app_shared_ptr,
+                                                policy_handler,
+                                                mock_application_manager));
+  }
+}
+
 TEST_F(MessageHelperTest, ProcessSoftButtons_Text_InvalidData) {
```

Comment:
@indlu I would like to propose you to divide code with empty lines into logically connected segments in order to improve readability. Some comments about the segments would be useful too.